kuma/wiki/models.py

```diff
@@ -1540,8 +1538,24 @@ def contributors(self):
         return DocumentContributorsJob().get(self.pk)
 
     @cached_property
-    def zone_stack(self):
-        return DocumentZoneStackJob().get(self.pk)
+    def nearest_zone(self):
+        """
+        The nearest zone for this document, starting from this document and
+        moving upwards via "parent_topic". For a non-default-language document
+        that does not have its own nearest zone, returns the nearest zone of
+        its parent (the document it was translated from).
+        """
+        job = DocumentNearestZoneJob()
+        result = job.get(self.pk)
+        if not result:
+            if (self.locale != settings.WIKI_DEFAULT_LANGUAGE) and self.parent:
```

Comment:
nitpick: this could be one long ugly ``if`` statement instead of a nested ``if``